agent/src/flow_generator/protocol_logs/mq/kafka.rs

```diff
@@ -352,22 +452,125 @@ impl KafkaLog {
         if proto != IpProtocol::TCP {
             return Err(Error::InvalidIpProtocol);
         }
-        if payload.len() < KAFKA_REQ_HEADER_LEN {
-            return Err(Error::KafkaLogParseFailed);
-        }
+
         match direction {
             PacketDirection::ClientToServer => {
+                if payload.len() < KAFKA_REQ_HEADER_LEN {
+                    return Err(Error::KafkaLogParseFailed);
+                }
                 self.request(payload, false, info)?;
                 self.perf_stats.as_mut().map(|p| p.inc_req());
             }
             PacketDirection::ServerToClient => {
+                if payload.len() < KAFKA_RESP_HEADER_LEN {
+                    return Err(Error::KafkaLogParseFailed);
+                }
                 self.response(payload, info)?;
                 self.perf_stats.as_mut().map(|p| p.inc_resp());
             }
         }
         Ok(())
     }
 
+    fn get_topics_name(
+        &mut self,
+        api_key: u16,
+        api_version: u16,
+        payload: &[u8],
+    ) -> Option<String> {
+        let mut _fixed_offset = kafka_apiversion_topic_fixed_offset!(api_key, api_version);
```

Comment:
What's the reason of using a `_name`?

Reply:
`kafka_apiversion_topic_fixed_offset` can return a `Option<usize>` to get rid of `usize::MAX`